src/hyp3_isce2/burst.py

```diff
@@ -0,0 +1,238 @@
+import copy
+import re
+import time
+import xml.etree.ElementTree as ET
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Iterator, List, Tuple, Union
+
+import pandas as pd
+import requests
+from shapely import geometry
+
+URL = 'https://sentinel1-burst.asf.alaska.edu'
+
+
+@dataclass
+class BurstParams:
+    """Class that contains the parameters nessecary to request a burst from the API."""
+
+    granule: str
+    swath: str
+    polarization: str
+    burst_number: int
+
+
+class BurstMetadata:
```

Comment:
I prefer to use meaningful variable names in list comprehensions rather than `x` and `y` for the sake of readability. Not necessarily blocking, but would be nice to have.